fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -1176,7 +1176,7 @@ protected void unprotectUpsertTransactionState(TransactionState transactionState
                     runningTxnNums++;
                 }
             }
-            if (transactionState.getTransactionStatus() == TransactionStatus.COMMITTED) {
+            if (Config.enable_new_publish_mechanism && transactionState.getTransactionStatus() == TransactionStatus.COMMITTED) {
```

Comment:
Should transactionGraph.remove dependent on this config?

Reply:
if it's empty, it wont have any effects